src/module/apps/dialogs/ShadowrunItemDialog.ts

```diff
@@ -95,22 +140,39 @@ export class ShadowrunItemDialog extends Dialog {
             },
         };
         dialogData.default = 'normal';
-        dialogData.close = async (html) => {
-            if (cancel) return false;
-            const force = Helpers.parseInputToNumber($(html).find('[name=force]').val());
-            await item.setLastSpellForce({ value: force, reckless });
-            return true;
+
+        return async (html: JQuery): Promise<ActionTestData|undefined> => {
+            if (cancel) return;
+
+            const actionTestData = {} as SpellActionTestData;
+
+            mergeObject(actionTestData, ShadowrunItemDialog._getSelectedSpellForce(html));
+            mergeObject(actionTestData, ShadowrunItemDialog._getSelectedSpellReckless(reckless));
+
+            await item.setLastSpellForce({value: actionTestData.force, reckless: actionTestData.reckless});
+
+            return {spell: actionTestData};
         };
     }
 
-    static addRangedWeaponData(templateData: object, dialogData: DialogData, item: SR5Item): void {
+    static _getSelectedSpellForce(html: JQuery): object {
```

Comment:
I like this design pattern.. I'll try to match it from now on

Reply:
I completly redid this at the current branch head with a custom FormDialog extension solving most of the promise logic / data flow inconsistencies. :)
This pattern is still there, just a hidden and now also easier to replace, should we decide to change the approach.